source/core.py

```diff
@@ -447,10 +447,10 @@ def handlePowerStatusChange(self):
 	if wxLang:
 		try:
 			locale.Init(wxLang.Language)
-			# Revert wx's changes to the python locale
-			languageHandler.setLocale(languageHandler.curLang)
 		except:
 			log.error("Failed to initialize wx locale",exc_info=True)
+		# Revert wx's changes to the python locale
+		languageHandler.setLocale(languageHandler.curLang)

```

Comment:
I guess you've moved this out of the try block because in the event that `locale.Init` fails NVDA should still try to set the thread, python, and languageHandler locale?

Reply:
`setLocale` just sets the python locale. But yes I figure it might be possible `locale.Init` fails after it sets the python locale to a breaking value so I moved it out of the except block. Might be worth putting it into a `finally:` block instead?

Reply:
Perhaps so, I guess that indicates more strongly that it is tied to the line inside the try block